azure-pipelines-dependency-graph-testing.yml

```diff
@@ -161,7 +161,11 @@ stages:
 
           - script: |
               pip install pytest pytest-cov pytest-azurepipelines
-              python scripts/determine_tests_to_run.py --depth 3 | xargs pytest $(GE_pytest_opts) --napoleon-docstrings --junitxml=junit/test-results.xml --cov=. --cov-report=xml --cov-report=html --ignore=tests/cli --ignore=tests/integration/usage_statistics
+              echo "Tests identified by dependency graph test runner [dgtr]:"
+              python scripts/determine_tests_to_run.py --depth 3 --ignore 'tests/cli' 'tests/integration/usage_statistics' | tee .dgtr
+              if [ -s .dgtr ]; then
+                cat .dgtr | xargs pytest $(GE_pytest_opts) --napoleon-docstrings --junitxml=junit/test-results.xml --cov=. --cov-report=xml --cov-report=html
+              fi
```

Comment:
Run the script and pipe the output to `tee`, which prints the result to STDOUT as well as to a file named `.dgtr`. This allows us to read the proposed test files in our Azure logs but also save the results for later.

`if [ -s .dgtr ]` checks if the file is not empty. This resolves an issue where no files would be outputted by the script, resulting in `pytest` assuming the whole suite should be run.

`cat` prints the file contents and `xargs` feeds it into `pytest`